test/e2e/app-dir/not-found/not-found.test.ts

```diff
@@ -18,6 +19,22 @@ createNextDescribe(
         expect(html).toContain("I'm still a valid page")
       })
 
+      if (isNextDev) {
+        it('should not reload the page', async () => {
```

Comment:
Can we add an additional test to ensure it does correctly reload when the non-existent path is added? Seems this case might need handling still